docs/template_plugin/src/template_executable_network.cpp

```diff
@@ -16,93 +16,85 @@
 #include <threading/ie_executor_manager.hpp>
 #include <details/ie_cnn_network_tools.h>
 
-#include <ngraph/ngraph.hpp>
-
-#include <transformations/common_optimizations/common_optimizations.hpp>
-
+#include "template/template_config.hpp"
 #include "template_plugin.hpp"
 #include "template_executable_network.hpp"
-#include "template_pattern_transformation.hpp"
 
 using namespace TemplatePlugin;
 
 // ! [executable_network:ctor_cnnnetwork]
-TemplatePlugin::ExecutableNetwork::ExecutableNetwork(InferenceEngine::ICNNNetwork&  network,
-                                                     const Configuration&           cfg):
-    _name(network.getName()),
+TemplatePlugin::ExecutableNetwork::ExecutableNetwork(const std::shared_ptr<ngraph::Function>&   function,
+                                                     const Configuration&                       cfg,
+                                                     const Plugin::Ptr&                         plugin) :
+    InferenceEngine::ExecutableNetworkThreadSafeDefault(nullptr, nullptr), // Disable default threads creation
     _cfg(cfg),
-    _waitExecutor(InferenceEngine::ExecutorManager::getInstance()->getExecutor("Template")) {
```

Comment:
why `_waitExecutor` is moved out of executable network?

Reply:
The Idea is to have one executor per device for all executable networks in plugin. So plugin is better place for it